apitest.js

```diff
@@ -516,4 +575,79 @@ describe('InternalFunctionCallTest1 Zapp', () => {
+  });
+
+  it('MinLeaf Index check', async () => {
+    expect(parseInt(res.Swap[0].body.tx.returnValues.minLeafIndex)).to.equal(0); // deposit 1
+    expect(parseInt(res.Swap[1].body.tx.returnValues.minLeafIndex)).to.equal(2); // deposit 2
+    expect(parseInt(res.Swap[3].body.tx.returnValues.minLeafIndex)).to.equal(4); // startSwap
+    expect(parseInt(res.Swap[5].body.tx.returnValues.minLeafIndex)).to.equal(8); // completeSwap
+  });
+
+  it('Check number of commitments', async () => {
+    expect(res.Swap[2].body.commitments.length).to.equal(4);
+    expect(res.Swap[4].body.commitments.length).to.equal(8);
+    expect(res.Swap[6].body.commitments.length).to.equal(14);
+  });
+
+  it('Check nullified commitments', async () => {
+    expect(res.Swap[2].body.commitments[0].isNullified).to.equal(false);
+    expect(res.Swap[2].body.commitments[1].isNullified).to.equal(false);
+    expect(res.Swap[2].body.commitments[2].isNullified).to.equal(false);
+    expect(res.Swap[2].body.commitments[3].isNullified).to.equal(false);
+
+    expect(res.Swap[4].body.commitments[0].isNullified).to.equal(true);
+    expect(res.Swap[4].body.commitments[1].isNullified).to.equal(true);
+    expect(res.Swap[4].body.commitments[2].isNullified).to.equal(true);
+    expect(res.Swap[4].body.commitments[3].isNullified).to.equal(false);
+    expect(res.Swap[4].body.commitments[4].isNullified).to.equal(false);
+    expect(res.Swap[4].body.commitments[5].isNullified).to.equal(false);
+    expect(res.Swap[4].body.commitments[6].isNullified).to.equal(false);
+    expect(res.Swap[4].body.commitments[7].isNullified).to.equal(false);
+
+    expect(res.Swap[6].body.commitments[0].isNullified).to.equal(true);
+    expect(res.Swap[6].body.commitments[1].isNullified).to.equal(true);
+    expect(res.Swap[6].body.commitments[2].isNullified).to.equal(true);
+    expect(res.Swap[6].body.commitments[3].isNullified).to.equal(true);
+    expect(res.Swap[6].body.commitments[4].isNullified).to.equal(false);
+    expect(res.Swap[6].body.commitments[5].isNullified).to.equal(true);
+    expect(res.Swap[6].body.commitments[6].isNullified).to.equal(true);
+    expect(res.Swap[6].body.commitments[7].isNullified).to.equal(true);
+    expect(res.Swap[6].body.commitments[8].isNullified).to.equal(false);
+    expect(res.Swap[6].body.commitments[9].isNullified).to.equal(false);
+    expect(res.Swap[6].body.commitments[10].isNullified).to.equal(false);
+    expect(res.Swap[6].body.commitments[11].isNullified).to.equal(false);
+    expect(res.Swap[6].body.commitments[12].isNullified).to.equal(false);
+    expect(res.Swap[6].body.commitments[13].isNullified).to.equal(false);
+  });
+
+  it('Check value of final commitment', async () => {
+    expect(parseInt(res.Swap[6].body.commitments[0].preimage.value)).to.equal(100); // deposit 1
+    expect(parseInt(res.Swap[6].body.commitments[2].preimage.value)).to.equal(100); // deposit 2
```

Comment:
Unrelated to this PR, but I don't understand why the balance isn't 200 after the second deposit of 100 tokens.

Reply:
Makes sense, thanks!